### PR TITLE
chore(frontend-canister): move asset-canister-related crates into SDK repo

### PR DESCRIPTION
**Pull request by smallstepman**
# Description

Move all asset-canister-related code into the SDK repo. This includes:
- `ic-asset`
- `icx-asset`
- `ic-certified-assets`
- `certified-assets` 

Closes [SDK-607](https://dfinity.atlassian.net/browse/SDK-607)

# How Has This Been Tested?

# Checklist:

- [x] The title of this PR complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] I have edited the CHANGELOG accordingly.
- [ ] I have made corresponding changes to the documentation.

### CODE REVIEW

File: .github/workflows/update-docs.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
           rustup default ${{ matrix.rust }}
 
       - name: Check cargo build
-        run: cargo build
+        run: cargo build
```
Comment (ericswanson-dfinity) [inline .github/workflows/update-docs.yml:25]:
```suggestion
        run: cargo build
```

**Review by ericswanson-dfinity — CHANGES_REQUESTED** [review]
Should this PR also update these crate dependencies?
- https://github.com/dfinity/sdk/blob/master/src/dfx/Cargo.toml#L90
- https://github.com/dfinity/sdk/blob/master/Cargo.toml#L11

**Comment by smallstepman** [comment]
@ericswanson-dfinity amazing as always to see you catch all of that

I had this change in other PR, but forgot to add it here

fixed

**Comment by smallstepman** [comment]
what do you think about dir name `src/assets_canister`? underscore vs hyphen? `assets` vs `asset`?

**Review by ericswanson-dfinity — CHANGES_REQUESTED** [review]
I notice this when I run `cargo clean` and so forth. 
```
$ cargo clean
warning: profiles for the non root package will be ignored, specify profiles at the workspace root:
package:   /Users/ericswanson/d/sdk/src/canisters/frontend/certified-assets/Cargo.toml
workspace: /Users/ericswanson/d/sdk/Cargo.toml
```

File: src/canisters/frontend/certified-assets/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+lto = true
+opt-level = 'z'
+panic = 'abort'
+
+[lib]
+path = "src/lib.rs"
+crate-type = ["cdylib"]
+
+[dependencies]
+ic-certified-assets = { path = "../ic-certified-assets", vesrion = "0.2.4" }
```
Comment (ericswanson-dfinity) [inline src/canisters/frontend/certified-assets/Cargo.toml:17]: typo
```suggestion
ic-certified-assets = { path = "../ic-certified-assets", version = "0.2.4" }
```

Comment (smallstepman) [reply src/canisters/frontend/certified-assets/Cargo.toml]: got rid of `version` altogether

**Comment by smallstepman** [comment]
@ericswanson-dfinity thanks, fixed it here https://github.com/dfinity/sdk/pull/2412/commits/676c2aac597ecee469eb5dba6cd545e80c479318

For some reason, it's not possible to specify `panic`/`lto`/`rpath` in root level Cargo package-specific profile overrides section
```
[profile.<name>.package.<name>]  # Override profile for a package.
# Same keys for a normal profile (minus `panic`, `lto`, and `rpath`).
```
so I had to make a global one (`[profile.release]`)

**Review by ericswanson-dfinity — APPROVED** [review]

**Pull request merged**